manifests/repo.pp

```diff
@@ -24,6 +24,17 @@
       sslcacert       => '/etc/pki/tls/certs/ca-bundle.crt',
       metadata_expire => 300,
     }
+    if versioncmp($repo_release, '8') >= 0 {
+      # This exec ensures GPG key is added without errors
+      # Initial GPG download for dnf appears to return no exit code, tries=2 is workaround
+      exec { 'dnf makecache sensu':
```

Comment:
Should you actually install the gpg key before the repo so this doesn't happen?

Reply:
This is easier. Installing the GPG key would require a lot more logic because we'd have to download the key then run RPM commands to import the key. Using this command auto-imports the key and is the same command used by upstream repo scripts from Package Cloud.

Reply:
Please add the explanation about how the upstream scripts do this as well instead of importing the gpg key.

Reply:
Explanation added with link to upstream page